platform/security/common/src/main/java/ddf/security/service/impl/AbstractAuthorizingRealm.java

```diff
@@ -56,25 +56,46 @@
 
     private static final String SAML_ROLE = "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/role";
 
-    protected Map<ServiceReference, Expansion> expansionServices = new ConcurrentHashMap<>();
+    protected Map<ServiceReference, Expansion> userExpansionServices = new ConcurrentHashMap<>();
 
-    public void addExpansion(ServiceReference<Expansion> expansionServiceRef) {
+    protected Map<ServiceReference, Expansion> metacardExpansionServices = new ConcurrentHashMap<>();
+
+    public void addUserExpansion(ServiceReference<Expansion> expansionServiceRef) {
+        Bundle bundle = FrameworkUtil.getBundle(AbstractAuthorizingRealm.class);
+        if (bundle != null) {
+            Expansion expansion = bundle.getBundleContext()
+                    .getService(expansionServiceRef);
+            addUserExpansion(expansionServiceRef, expansion);
+        }
+    }
+
+    public void addUserExpansion(ServiceReference<Expansion> expansionServiceRef, Expansion expansion) {
+        if (expansionServiceRef != null) {
+            userExpansionServices.put(expansionServiceRef, expansion);
```

Comment:
I was actually referring to the value. We're guarding the key but allowing a null value to be inserted.